packages/terra-i18n-plugin/src/I18nAggregatorPlugin.js

```diff
@@ -15,42 +14,48 @@ const messages = ${JSON.stringify(messages, null, 2)};
 module.exports = {
   areTranslationsLoaded: true,
   locale: '${language}',
-  messages,
+  messages: messages
 };`;
 }
 
-function getDirectories(srcPath) {
-  return fs.readdirSync(srcPath).filter(file => fs.statSync(path.join(srcPath, file)).isDirectory());
+function getDirectories(srcPath, compiler) {
+  return compiler.inputFileSystem.readdirSync(srcPath).filter(file => compiler.inputFileSystem.statSync(path.join(srcPath, file)).isDirectory());
 }
 
-function aggregateDirectory(languageMessages, currentDirectory) {
+function aggregateDirectory(languageMessages, currentDirectory, compiler) {
   // Check the directory for translations
   const translationsDirectory = path.resolve(currentDirectory, 'translations');
-  if (fs.existsSync(translationsDirectory)) {
+  try {
+    // Check if the directory exists by attempting to read from it
+    compiler.inputFileSystem.readdirSync(translationsDirectory);
+
     // Check the directory for each translation file
     supportedLocales.forEach((language) => {
       const translationFile = path.resolve(translationsDirectory, `${language}.json`);
-      if (fs.existsSync(translationFile)) {
-        Object.assign(languageMessages[language], JSON.parse(fs.readFileSync(translationFile, 'utf8')));
-      } else {
-        // eslint-disable-next-line no-console
+      try {
+        Object.assign(languageMessages[language], JSON.parse(compiler.inputFileSystem.readFileSync(translationFile, 'utf8')));
+      } catch (e) {
         console.warn(`Translation file ${language}.json not found for ${translationsDirectory}`);
       }
     });
+  } catch (e) {
+    console.log(e.message);
```

Comment:
the linter would not allow me to have no catch block or an empty catch block; previously we were not logging out anything here, but I put in the log as nothing else really to do here to get the linter error from occurring.  This causes many many logs to appear when starting the webpack-dev-server that previously did not appear.  It also seems to be causing the Travis CI job to fail as the output becomes too large.  Therefore, should we allow the linter to ignore not having a catch block or possibly do something else here?

Reply:
For reference the existsSync call was removed because the default input file system doesn't support it. I believe webpack itself just try-catches file reads. I'm ok with disabling the linter in this specific plugin instance.